<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Carson Cook</title>
    <!-- Include Bootstrap CSS -->
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css">
    <!-- Link to external CSS file -->
    <link rel="stylesheet" href="styles.css">
    <!-- Other head content -->
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css">
</head>
<body>
    <div class="header" id="header">
        <h1>Carson Cook</h1>
        <p>Data Science | ML Engineering</p>
        <div class="hamburger" id="hamburger">
            <div>
                <div class="hamburger-line" id="line-1"></div>
                <div class="hamburger-line" id="line-2"></div>
                <div class="hamburger-line" id="line-3"></div>
            </div>
        </div>
        
        <ul id="menu">
            <li><a href="#summary">Summary</a></li>
            <li><a href="#research">Research</a></li>
            <li><a href="#personal-projects">Personal Projects</a></li>
            <li><a href="#education">Education</a></li>
            <li><a href="#skills">Skills</a></li>
            <li><a href="#resume-and-links">Resume and Links</a></li>
        </ul>
    </div>
    <!-- Summary Section -->
    <section id="summary" class="content-section">
        <!-- <h2>Summary</h2> -->
        <div class="summary-container">
            <img src="media/headshot.png" id="headshot" alt="Headshot">
            <p class="subheading">I am a machine learning and data science practitioner who is passionate about tackling challenging problems in applied areas.</p>
            
        </div>
        <hr class="subsection-divider">
        
        <p>
            My academic and professional career have both involved applications of <strong>machine learning, deep learning, and statistical modeling</strong>. In my professional career, I currently focus on <strong>latent knowledge state modeling, content recommendation systems, and natural language processing</strong>. Previously, I worked on projects related to scientific instrumentation, including cell image analysis with deep learning and flow cytometry data analysis and collection. I also have professional experience in traditional software engineering and data analysis. In my graduate research, I focused primarily on transcriptomic analysis and image modeling for applications in tissue engineering.
        </p>

        <hr class="section-divider">
    </section>

    <!-- Research Section -->
    <section id="research" class="content-section">
        <h2>Research</h2>
        <p class="subheading">A selection of my academic and professional research.</p>

        <hr class="subsection-divider">
        <h3>LENS: Predictive Diagnostics for Flexible and Efficient Assessments</h3>
        <p>
            S. Thomas Christie, Hayden Johnson, Carson Cook, Garron Gianopulos, Anna Rafferty. <a href="https://doi.org/10.1145/3573051.3593392" target="_blank">Published: Proceedings from L@S (2023).</a>
            <blockquote>
                The utility of assessment systems lies in their capacity to <strong>transform observations of student behavior into meaningful inferences about learning, knowledge, and skills</strong>. Common practice is to use latent variable models and produce scores on scales. However the simplicity of these psychometric models may filter out potentially valuable information present in student behavior. In particular, scale scores are not optimized to support granular instructional decisions. Machine learning offers promising alternatives, but proposed deep learning architectures are not ideally suited for operational testing conditions involving sparse data and shifting category labels for test questions.
                
                We present criteria for a <strong>model architecture that can leverage the rich, sparse behavioral data made available by modern assessment systems</strong>. We argue that behavioral predictions with well-quantified uncertainties are a viable alternative to scale scores for many applications. To satisfy these goals, we propose the LENS model architecture that combines the flexibility of machine learning and the uncertainty quantification of latent variable models to produce high-quality predictive diagnostic claims. <strong>Inspired by variational autoencoders, LENS maps student behavior to a latent probability distribution</strong>. In addition, LENS performs explicit <strong>Bayesian integration of each behavioral observation, allowing the model to incorporate sparse data with informative prior beliefs about students.</strong>
                
                We then compare the predictive capability of LENS to both latent variable and machine learning approaches. <strong>LENS generates competitive or superior predictions of behavior, particularly on sparse data, while requiring fewer modeling assumptions than other options</strong> and allowing for easy incorporation of auxiliary behavioral data. We also show how to produce interpretable claims about student behavior at multiple levels of granularity, allowing the same model to serve multiple reporting needs. Finally, we discuss the flexibility afforded by the proposed model for constructing assessments customized to both educator needs and student skills.
            </blockquote>
        </p>
        <div class="flex-container">
            <a href="https://doi.org/10.1145/3573051.3593392" target="_blank"  id="publication">
                <img src="media/lens_paper_figure.png" class="responsive-image">
            </a>
        </div>
        <hr class="subsection-divider">

        <h3>Student State Space Modeling with Bayesian Information Integration</h3>
        <p>
            Carson Cook, Garron Gianopulos, Emily Lubkert, Yeow Meng Thum, and S. Thomas Christie. <a href="media/bayesian_information_integration_for_summative_scores_presentation.pdf" target="_blank">Presented: NCME Annual Meeting (2023), NCSA (2023).</a>
        </p>
        <blockquote>
            In the United States, summative assessments - <strong>standardized tests taken at a single point in time, typically at the end of the school year - are widely used for accountability purposes; however, the results of these assessments come too late in the school year to have an impact on classroom instruction.</strong> In response, many districts look to interim assessments, which are administered multiple times through the school year, for more timely feedback on the student progress. Through-year assessments are an emerging alternative where the goal is frequently to administer interim assessments and use their results to arrive at summative scores. <strong>This approach is intended to satisfy the twin objectives of (1) providing timely feedback on student progress and (2) eliminate the need for a separate summative test to reduce testing burden.</strong> Unfortunately, this creates another challenge: how should multiple test scores be combined to produce a single summative score? If the goal is to arrive at a summative score that combines information across tests, is measurement-centric, and requires few hand-tuning decisions, there is currently no favored method.
            
            To address this challenge, <strong>we created a Bayesian state space model, inspired by the Kalman filter, that combines information across test events to arrive at a composite score that can be used for summative purposes.</strong> Our model uses three test scores - fall, winter, and spring - together with expected growth to produce an estimate of student proficiency in the spring. We fit our model to simulated data and found that, though the estimates produced by our model were somewhat biased for simulated students with atypical growth, it performed substantially better in terms of RMSE than the using the spring score alone. We also evaluated the model fit on a historical MAP Growth dataset, via Bayesian model comparison, and found that the proposed model is more likely to have generated the data than the baseline model, which simply uses spring scores as the ability estimates.
        </blockquote>
        <div class="flex-container">
            <a href="media/bayesian_information_integration_for_summative_scores_presentation.pdf" target="_blank"  id="presentation">
                <img src="media/bayesian_state_space_model.png" class="responsive-image">
            </a>
        </div>
        <hr class="subsection-divider">

        <h3>Group-Level Ability Estimation Using Bayesian Multilevel Modeling: Opportunities and Limitations</h3>
        <p>
            S. Thomas Christie, Carson Cook, and Garron Gianopulos. <a href="media/OTL_overview.pdf" target="_blank">Presented: NCME Annual Meeting (2023), NCSA (2023).</a>
        </p>
        <blockquote>
            Granular, <strong>instructionally-relevant sub-scores have the potential to increase the utility of standardized assessment reports for teachers in the classroom.</strong> However, existing procedures tend to produce sub-scores with poor reliability due to the small number of item responses used to compute them. <strong>Since instructional planning decisions are often made for groups of students, we introduce and evaluate a model for group-level sub-scores that allows us to pool information across students and produces one sub-score per reporting category per student group.</strong> Our primary interest is whether coordinate shift in performance by students is detectable given our method. Specifically, we investigate whether a granular class-level score reveals a detectable difference in performance within specific reporting categories when compared to the responses predicted by overall scores. Through this lens, a sub-score would identify class-wide areas of strength or opportunities for class-level instruction, thereby assisting teachers with instructional planning.
            
            To validate the proposed approach, <strong>we conducted a simulation study and determined the effect size, class size, and the number of item responses required to detect class-level performance differences within granular item reporting categories.</strong> We then applied our modeling approach to historical ELA and Math assessment data to characterize its potential utility in those domains. Our findings indicate that this method may have utility for identifying variation in class-level performance at the level of reporting categories, but this is highly dependent on the test blueprint and the magnitude of the effect.
        </blockquote>
        <div class="flex-container">
            <a href="media/OTL_overview.pdf" target="_blank"  id="presentation">
                <img src="media/group_plate_model_diagram.png" class="responsive-image">
            </a>
        </div>
        <hr class="subsection-divider">

        <h3>Characterizing the Extracellular Matrix Transcriptome of Endometriosis</h3>
        <p>
            Carson Cook, Noah Wiggin, and Kaitlin C. Fogg. <a href="https://doi.org/10.1007/s43032-023-01359-w" target="_blank">Published: Reproductive Sciences (2023).</a>
            <blockquote>
                In recent years, <strong>the matrisome, a set of proteins that make up the extracellular matrix (ECM) or are closely involved in ECM behavior, has been shown to have great importance for characterizing and understanding disease pathogenesis and progression.</strong> The matrisome is especially critical for examining diseases characterized by extensive tissue remodeling. <strong>Endometriosis is characterized by the extrauterine growth of endometrial tissue, making it an ideal condition to study through the lens of matrisome gene expression.</strong> While large gene expression datasets have become more available and gene dysregulation in endometriosis has been the target of several studies, the gene expression profile of the matrisome specifically in endometriosis has not been well characterized. In our study, we explored four Gene Expression Omnibus (GEO) DNA microarray datasets containing eutopic endometrium of people with and without endometriosis. After batch correction, menstrual cycle phase accounted for 53% of variance and disease accounted for 23%; thus, the data were separated by menstrual cycle phase before performing <strong>differential expression analysis, statistical and machine learning modeling, and enrichment analysis. We established that matrisome gene expression alone can effectively differentiate endometriosis samples from healthy ones, demonstrating the potential of matrisome gene expression for diagnostic applications. Furthermore, we identified specific matrisome genes and gene networks whose expression can distinguish endometriosis stages I/II from III/IV.</strong> Taken together, these findings may aid in developing future in vitro models of disease, offer insights into novel treatment strategies, and advance diagnostic tools for this underserved patient population.
            </blockquote>
        </p>
        <div class="flex-container">
            <a href="https://doi.org/10.1007/s43032-023-01359-w" target="_blank"  id="publication">
                <img src="media/endometriosis_paper_figure.png" class="responsive-image">
            </a>
        </div>
        <hr class="subsection-divider">

        <h3>Characterizing the extracellular matrix transcriptome of cervical, endometrial, and uterine cancers</h3>
        <p>
            Carson Cook, Andrew Miller, Thomas Barker, Yanming Di, Kaitlin Fogg. <a href="https://doi.org/10.1016/j.mbplus.2022.100117">Published: Matrix Biology Plus (2022).</a>
            <blockquote>
                Increasingly, <strong>the matrisome, a set of proteins that form the core of the extracellular matrix (ECM) or are closely associated with it, has been demonstrated to play a key role in tumor progression.</strong> However, in the context of gynecological cancers, the matrisome has not been well characterized. <strong>A holistic, yet targeted, exploration of the tumor microenvironment is critical for better understanding the progression of gynecological cancers, identifying key biomarkers for cancer progression, establishing the role of gene expression in patient survival, and for assisting in the development of new targeted therapies.</strong> In this work, we explored the matrisome gene expression profiles of cervical squamous cell carcinoma and endocervical adenocarcinoma (CESC), uterine corpus endometrial carcinoma (UCEC), and uterine carcinosarcoma (UCS) using publicly available RNA-seq data from The Cancer Genome Atlas (TCGA) and The Genotype-Tissue Expression (GTEx) portal. We hypothesized that the matrisomal expression patterns of CESC, UCEC, and UCS would be highly distinct with respect to genes which are differentially expressed and hold inferential significance with respect to tumor progression, patient survival, or both. <strong>Through a combination of statistical and machine learning analysis techniques, we identified sets of genes and gene networks which characterized each of the gynecological cancer cohorts.</strong> Our findings demonstrate that the matrisome is critical for characterizing gynecological cancers and transcriptomic mechanisms of cancer progression and outcome. Furthermore, while the goal of pan-cancer transcriptional analyses is often to highlight the shared attributes of these cancer types, we demonstrate that they are highly distinct diseases which require separate analysis, modeling, and treatment approaches. In future studies, matrisome genes and gene ontology terms that were identified as holding inferential significance for cancer stage and patient survival can be evaluated as potential drug targets and incorporated into in vitro models of disease.
            </blockquote>
        </p>
        <div class="flex-container">
            <a href="https://doi.org/10.1016/j.mbplus.2022.100117" target="_blank"  id="publication">
                <img src="media/gyn_cancer_paper_figure.png" class="responsive-image">
            </a>
        </div>
        <hr class="section-divider">
        
    </section>

    <!-- Personal Projects Section -->
    <section id="personal-projects" class="content-section">
        <h2>Personal Projects</h2>
        <p class="subheading">A selection of personal projects demonstrating my expertise in machine learning and data science.</p>
        <hr class="subsection-divider">
        <h3>Predicting Pitchfork Album Ratings from Review Text</h3>
        <h4>Goal</h4>
        <p>
            This project provides a start-to-finish example of an NLP transfer learning task, detailing data cleaning, exploratory data analysis, model fitting, and post-fit model prediction analysis.
        </p>
        <h4>Summary</h4>
        <p>
            In this project, I built a regression model to predict album ratings based on review text body. This was accomplished by building a regressor on top of a pre-trained transformer model and performing task-specific fine-tuning. This sequence of notebooks utilizes a <a href="https://pitchfork.com/" target="_blank">Pitchfork</a> reviews dataset of approximately 20K album reviews (<a href="https://huggingface.co/datasets/mattismegevand/pitchfork" target="_blank">mattismegevand/pitchfork</a>). The project consists of a sequence of jupyter notebooks and a corresponding custom library. The notebooks are available as both the original <code>.ipynb</code> source and <code>.html</code> exports.
        </p>
        <div class="flex-container" id="repo">
            <a href="https://github.com/cjc77/nlp-projects#pitchfork-rating-prediction" target="_blank">
                <img src="media/pitchfork_rating_distribution.png" class="responsive-image">
            </a>
        </div>
        <h4>Links</h4>
        <ul>
            <li><a href="https://github.com/cjc77/nlp-projects#pitchfork-rating-prediction" target="_blank">README (GitHub)</a></li>
            <li><a href="https://github.com/cjc77/nlp-projects/tree/main/notebooks_pitchfork_ratings" target="_blank">Notebooks (GitHub, .ipynb)</a></li>
            <li><a href="./notebooks_pitchfork_ratings/01_initial_data_prep.html" target="_blank">Notebooks (.html)</a></li>
        </ul>
        <hr class="subsection-divider">
        <h3>Exploring Pooling in Bayesian Linear Modeling</h3>
        <p>
            Coming soon...
        </p>
        <hr class="section-divider">
    </section>

    <!-- Education Section -->
    <section id="education" class="content-section">
        <h2>Education</h2>
        <h3>M.S. Bioengineering (M), Statistics (m), Oregon State University (2022)</h3>
        <p>
            Focus on computational transcriptomics and image analysis using statistical inference, machine learning, and deep learning.
        </p>
        <p>
            Advisors: <a href="https://engineering.oregonstate.edu/people/kaitlin-fogg" target="_blank">Kaitlin Fogg</a> (M), <a href="https://stat.oregonstate.edu/directory/yanming-di" target="_blank">Yanming Di</a> (m)
            <br>
            Lab: <a href="https://sites.google.com/oregonstate.edu/fogglab/research" target="_blank">The Fogg Lab</a>
        </p>
        <!-- <hr class="subsection-divider"> -->
        <h3>B.S. Computer Science (M), Mathematics (m), Portland State University (2020)</h3>
        <p>
            Focus on machine learning and data science. Participated in <a href="https://www.mecopinc.org/" target="_blank">MECOP (Multiple Engineering Cooperative Program)</a>.
        </p>
        <hr class="section-divider">
    </section>

    <!-- Skills Section -->
    <section id="skills" class="content-section">
        <h2>Skills</h2>
        <div class="flex-container">
            <div class="column">
                <h3>Machine Learning</h3>
                <p>
                    Over 4 years of hands-on machine learning experience in industry and academia, along with formal education in the subject. Able to break down domain-specific challenges into machine learning tasks and apply state-of-the-art modeling approaches. Experienced with a broad range of tasks and topics, including sequence modeling, representation learning, natural language processing, and image processing.
                </p>
            </div>
            <div class="column">
                <h3>Statistical Modeling & Data Analysis</h3>
                <p>
                    Over 5 years of hands-on data analysis experience in industry and academia, along with formal education in statistics and mathematics (including: mathematical statistics, statistical learning, calculus, differential equations, and other topics). Experience building custom Bayesian models in Stan, performing general statistical analysis and modeling using the R and python statistics/data science ecosystems, and conducting specialized bioinformatics analysis and inference using the Bioconductor ecosystem.
                </p>
            </div>
        </div>
        <div class="flex-container">
            <div class="column">
                <h3>Software Engineering & ML Ops</h3>
                <p>
                    Over 5 years of hands-on experience writing software in various languages (E.g., python, R, C#) for a wide variety of use-cases. Fluency with common SWE tools like Git and GitHub. Experience with cloud computing platforms like AWS SageMaker for model development and training pipelines.
                </p>
            </div>
            <div class="column">
                <h3>Communication</h3>
                <p>
                    Experienced in written and verbal technical communication, both in academic and industry settings. Able to engage with interdisciplinary teams, distill complex concepts into interpretable pieces, and able to author and contribute to technical documentation and research papers.
                </p>
            </div>
        </div>
          
        <hr class="section-divider">
    </section>

    <!-- Resume Section -->
    <section id="resume-and-links" class="content-section" style="margin-bottom: 60px;">
        <h2>Resume and Links</h2>
        <p style="margin-top: 20px;">
            <a href="https://docs.google.com/document/d/1iJR1QjKW_osyzJVPYxgNvjlQ4CO5oF9oTSSI6ZybpsY/edit?usp=sharing" target="_blank" >
                <i class="fa fa-file"></i> Resume (Google Doc)
            </a>
            <br>
            <a href="https://www.linkedin.com/in/carson-cook-15269a32" target="_blank">
                <i class="fab fa-linkedin"></i> LinkedIn
            </a>
            <br>
            <a href="https://github.com/cjc77" target="_blank">
                <i class="fab fa-github"></i> GitHub
            </a>
        </p>
    </section>

    <div class="footer">
        <div class="icon-container">
            <a href="https://www.linkedin.com/in/carson-cook-15269a32" target="_blank" class="footer-icon">
                <i class="fab fa-linkedin fa-2x"></i>
            </a>
            <a href="https://github.com/cjc77" target="_blank" class="footer-icon">
                <i class="fab fa-github fa-2x"></i>
            </a>
        </div>
        <span class="copyright">© Carson Cook </span>
    </div>
      
    <!-- Include Bootstrap JS (Optional) -->
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"></script>
    <script src="script.js"></script>
</body>
</html>
