<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Table of Contents</title>
    <!-- Include Bootstrap CSS -->
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css">
    <!-- Link to external CSS file -->
    <link rel="stylesheet" href="styles.css">
    <!-- Other head content -->
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.0.0-beta3/css/all.min.css">
</head>
<body>
    <div class="header">
        <h1>Carson Cook</h1>
        <div class="hamburger" id="hamburger">
            <div>
                <div class="hamburger-line"></div>
                <div class="hamburger-line"></div>
                <div class="hamburger-line"></div>
            </div>
        </div>
        
        <p>Data Science | ML Engineering</p>
        <ul id="menu">
            <li><a href="#summary">Summary</a></li>
            <li><a href="#portfolio">Portfolio</a></li>
            <li><a href="#research">Research</a></li>
            <li><a href="#education">Education</a></li>
            <li><a href="#skills">Skills</a></li>
            <li><a href="#resume-and-links">Resume and Links</a></li>
        </ul>
    </div>
    <!-- Summary Section -->
    <section id="summary" class="content-section">
        <h2>Summary</h2>
        <p class="subheading">I am a machine learning and data science practitioner who is passionate about tackling challenging problems in applied areas.</p>
        <p>
            My academic and professional career have both involved applications of <strong>machine learning, deep learning, and statistical modeling</strong>. In my professional career, I currently focus on <strong>latent knowledge state modeling, content recommendation systems, and natural language processing</strong>. Previously, I worked on projects related to scientific instrumentation, including cell image analysis with deep learning and flow cytometry data analysis and collection. I also have professional experience in traditional software engineering and data analysis. In my graduate research, I focused primarily on transcriptomic analysis and image modeling for applications in tissue engineering.
        </p>

        <hr class="section-divider">
    </section>

    <!-- Portfolio Section -->
    <section id="portfolio" class="content-section">
        <h2>Portfolio</h2>
        <p class="subheading">A selection of personal projects demonstrating my expertise in machine learning and data science.</p>
        <hr class="subsection-divider">
        <h3>Predicting Pitchfork Album Ratings from Review Text</h3>
        <h4>Goal</h4>
        <p>
            The intended purpose of this project was to provide a start-to-finish example of an NLP transfer learning task, detailing data cleaning, exploratory data analysis, model fitting, and post-fit model prediction analysis.
        </p>
        <h4>Summary</h4>
        <p>
            In this project, I built a regression model to predict album ratings based on review text body. This was accomplished by building a regressor on top of a pre-trained transformer model and performing task-specific fine-tuning. This sequence of notebooks utilizes a <a href="https://pitchfork.com/">Pitchfork</a> reviews dataset of approximately 20K album reviews (<a href="https://huggingface.co/datasets/mattismegevand/pitchfork">mattismegevand/pitchfork</a>). The project consists of a sequence of jupyter notebooks and a corresponding custom library. The notebooks are available as both the original <code>.ipynb</code> source and <code>.html</code> exports.
        </p>
        <h4>Links</h4>
        <ul>
            <li><a href="https://github.com/cjc77/nlp-projects#pitchfork-rating-prediction" target="_blank">README (GitHub)</a></li>
            <li><a href="https://github.com/cjc77/nlp-projects/tree/main/notebooks_pitchfork_ratings" target="_blank">Notebooks (GitHub, .ipynb)</a></li>
            <li><a href="./notebooks_pitchfork_ratings/01_initial_data_prep.html" target="_blank">Notebooks (.html)</a></li>
        </ul>
        <hr class="subsection-divider">
        <h3>Exploring Pooling in Bayesian Linear Modeling</h3>
        <p>
            Coming soon...
        </p>
        <hr class="section-divider">
    </section>
    <!-- Research Section -->
    <section id="research" class="content-section">
        <h2>Research</h2>
        <p class="subheading">A selection of my academic and professional research.</p>

        <hr class="subsection-divider">
        <h3>LENS: Predictive Diagnostics for Flexible and Efficient Assessments</h3>
        <p>
            <a href="https://doi.org/10.1145/3573051.3593392" target="_blank">S. Thomas Christie, Hayden Johnson, Carson Cook, Garron Gianopulos, Anna Rafferty. Published: Proceedings from L@S (2023).</a>
            <blockquote>
                The utility of assessment systems lies in their capacity to transform observations of student behavior into meaningful inferences about learning, knowledge, and skills. Common practice is to use latent variable models and produce scores on scales. However the simplicity of these psychometric models may filter out potentially valuable information present in student behavior. In particular, scale scores are not optimized to support granular instructional decisions. Machine learning offers promising alternatives, but proposed deep learning architectures are not ideally suited for operational testing conditions involving sparse data and shifting category labels for test questions.

                We present criteria for a model architecture that can leverage the rich, sparse behavioral data made available by modern assessment systems. We argue that behavioral predictions with well-quantified uncertainties are a viable alternative to scale scores for many applications. To satisfy these goals, we propose the LENS model architecture that combines the flexibility of machine learning and the uncertainty quantification of latent variable models to produce high-quality predictive diagnostic claims. Inspired by variational autoencoders, LENS maps student behavior to a latent probability distribution. In addition, LENS performs explicit Bayesian integration of each behavioral observation, allowing the model to incorporate sparse data with informative prior beliefs about students.
                
                We then compare the predictive capability of LENS to both latent variable and machine learning approaches. LENS generates competitive or superior predictions of behavior, particularly on sparse data, while requiring fewer modeling assumptions than other options and allowing for easy incorporation of auxiliary behavioral data. We also show how to produce interpretable claims about student behavior at multiple levels of granularity, allowing the same model to serve multiple reporting needs. Finally, we discuss the flexibility afforded by the proposed model for constructing assessments customized to both educator needs and student skills.
            </blockquote>
        </p>

        <hr class="subsection-divider">

        <h3>Characterizing the Extracellular Matrix Transcriptome of Endometriosis</h3>
        <p>
            <a href="https://doi.org/10.1007/s43032-023-01359-w" target="_blank"> Carson Cook, Noah Wiggin, and Kaitlin C. Fogg. Published: Reproductive Sciences (2023).</a>
            <blockquote>
                In recent years, the matrisome, a set of proteins that make up the extracellular matrix (ECM) or are closely involved in ECM behavior, has been shown to have great importance for characterizing and understanding disease pathogenesis and progression. The matrisome is especially critical for examining diseases characterized by extensive tissue remodeling. Endometriosis is characterized by the extrauterine growth of endometrial tissue, making it an ideal condition to study through the lens of matrisome gene expression. While large gene expression datasets have become more available and gene dysregulation in endometriosis has been the target of several studies, the gene expression profile of the matrisome specifically in endometriosis has not been well characterized. In our study, we explored four Gene Expression Omnibus (GEO) DNA microarray datasets containing eutopic endometrium of people with and without endometriosis. After batch correction, menstrual cycle phase accounted for 53% of variance and disease accounted for 23%; thus, the data were separated by menstrual cycle phase before performing differential expression analysis, statistical and machine learning modeling, and enrichment analysis. We established that matrisome gene expression alone can effectively differentiate endometriosis samples from healthy ones, demonstrating the potential of matrisome gene expression for diagnostic applications. Furthermore, we identified specific matrisome genes and gene networks whose expression can distinguish endometriosis stages I/II from III/IV. Taken together, these findings may aid in developing future in vitro models of disease, offer insights into novel treatment strategies, and advance diagnostic tools for this underserved patient population.
            </blockquote>
        </p>
        <hr class="section-divider">
    </section>

    <!-- Education Section -->
    <section id="education" class="content-section">
        <h2>Education</h2>
        <h3>M.S. Bioengineering (M), Statistics (m), Oregon State University (2022)</h3>
        <p>
            Focus on computational transcriptomics and image analysis using statistical inference, machine learning, and deep learning.
        </p>
        <p>
            Advisors: <a href="https://engineering.oregonstate.edu/people/kaitlin-fogg" target="_blank">Kaitlin Fogg</a> (M), <a href="https://stat.oregonstate.edu/directory/yanming-di" target="_blank">Yanming Di</a> (m)
            <br>
            Lab: <a href="https://sites.google.com/oregonstate.edu/fogglab/research" target="_blank">The Fogg Lab</a>
        </p>
        <!-- <hr class="subsection-divider"> -->
        <h3>B.S. Computer Science (M), Mathematics (m), Portland State University (2020)</h3>
        <p>
            Focus on machine learning and data science. Participated in <a href="https://www.mecopinc.org/" target="_blank">MECOP (Multiple Engineering Cooperative Program)</a>.
        </p>
        <hr class="section-divider">
    </section>

    <!-- Skills Section -->
    <section id="skills" class="content-section">
        <h2>Skills</h2>
        <div class="flex-container">
            <div class="column">
                <h3>Machine Learning</h3>
                <p>
                    Over 4 years of hands-on machine learning experience in industry and academia, along with formal education in the subject. Able to break down domain-specific challenges into machine learning tasks and apply state-of-the-art modeling approaches. Experienced with a broad range of tasks and topics, including sequence modeling, representation learning, natural language processing, and image processing.
                </p>
            </div>
            <div class="column">
                <h3>Statistical Modeling & Data Analysis</h3>
                <p>
                    Over 4 years of hands-on data analysis experience in industry and academia, along with formal education in statistics and mathematics (including: mathematical statistics, statistical learning, calculus, differential equations, and other topics). Experience building custom Bayesian models in Stan, performing general statistical analysis and modeling using the R and python statistics/data science ecosystems, and conducting specialized bioinformatics analysis and inference using the Bioconductor ecosystem.
                </p>
            </div>
            <div class="column">
                <h3>Communication</h3>
                <p>
                    Experienced in written and verbal technical communication, both in academic and industry settings. Able to engage with interdisciplinary teams, distill complex concepts into interpretable pieces, and able to author and contribute to technical documentation and research papers.
                </p>
            </div>
            
        </div>
          
        <hr class="section-divider">
    </section>

    <!-- Resume Section -->
    <section id="resume-and-links" class="content-section" style="margin-bottom: 60px;">
        <h2>Resume and Links</h2>
        <p style="margin-top: 20px;">
            <a href="https://docs.google.com/document/d/1iJR1QjKW_osyzJVPYxgNvjlQ4CO5oF9oTSSI6ZybpsY/edit?usp=sharing" target="_blank" >
                <i class="fa fa-file"></i> Resume (Google Doc)
            </a>
            <br>
            <a href="https://www.linkedin.com/in/carson-cook-15269a32/" target="_blank">
                <i class="fab fa-linkedin"></i> LinkedIn
            </a>
            <br>
            <a href="https://github.com/cjc77" target="_blank">
                <i class="fab fa-github"></i> GitHub
            </a>
        </p>
    </section>

    <div class="footer">
        <div class="icon-container">
          <a href="https://github.com/your-username" target="_blank" class="footer-icon">
            <i class="fab fa-github fa-2x"></i>
          </a>
          <a href="https://www.linkedin.com/in/your-profile/" target="_blank" class="footer-icon">
            <i class="fab fa-linkedin fa-2x"></i>
          </a>
        </div>
        <span class="copyright">© Carson Cook </span>
      </div>
      
    <!-- Include Bootstrap JS (Optional) -->
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js"></script>
    <script src="script.js"></script>
</body>
</html>
